rfd/0173-terraform-machine-id.md

```diff
@@ -0,0 +1,303 @@
+
+Reusing the same resource (delete/re-create) proved to be very harmful when we did this in the Kubernetes operator.
+This caused a lot of instability/consistency issues, it took a full operator rewrite to solve them.
+
+If needed we can list how many terraform bot resources are living in Teleport and warn the user if it goes above a
+certain threshold, but this should not be necessary.
+
+The issue caused by the number of resources will very likely be addressed in the future by the work done on
+[the `BotInstance` resource](https://github.com/gravitational/teleport/pull/36510).
+This will allow tokens (even secret ones) and bots to be shared by multiple instances.
+
+### Privacy
+
+The fact a user ran a Terraform command with `bootstrap` is disclosed via the bot resource.
+Once could infer when Terraform commands are executed from this. This information is already available to admins via the
+audit log.
+
+### UX
+
+This improves the UX for the "existing advanced user" persona as they don't need to install and configure `tbot`
+anymore. This also unblocks support for runtimes where running `tbot` was not possible:
+e.g. [Terraform Cloud](https://github.com/gravitational/teleport/issues/26345).
+
+This greatly improves the UX of the "getting started" personas as the provider will hide all the complexity and shorten
+the time to value for IaC adoption. The
+whole [setup Terraform provider page](https://github.com/gravitational/teleport/blob/master/docs/pages/management/dynamic-resources/terraform-provider.mdx)
+becomes a 2-step guide: run `tsh login` and create the `main.tf`.
+
+Two documentation guides will be published:
+- Getting started with the Terraform Provider, explaining users how to run the provider locally
+- Running the Terraform Provider in CI/CD pipelines, explaining users how to run the provider in GitHub Actions, GitLab
+  CI, Circle CI, Spacelift, (and Terraform Cloud when we'll add support).
+
+The Getting started guide should provide instruction on how to connect to a local cluster with self-signed certificates
+(recover and trust the cluster cert with `SSL_CERT_FILE` instead of running with --insecure).
+
+### Observability
+
+The new Provider joining mechanisms will reuse existing Teleport APIs which are already emitting audit events and
+reporting Prometheus metrics.
+
+By its short-lived nature, the terraform process does not expose metrics.
+
+### Product usage
+
+To validate product adoption and join method usage we need some Telemetry, there are two possible approaches:
+
+#### Anonymous opt-in Telemetry client-side
+
+In this approach, telemetry would be opt-in and reuse the existing tbot start event and adding new fields:
+
+```protobuf
+syntax = "proto3";
+
+message TbotStartEvent {
+  enum RunMode {
+    RUN_MODE_UNSPECIFIED = 0;
+    RUN_MODE_ONE_SHOT = 1;
+    RUN_MODE_DAEMON = 2;
+    RUN_MODE_IN_PROCESS = 3;
+  }
+  RunMode run_mode = 1;
+  string version = 2;
+  string join_type = 3;
+  string helper = 4;
```

Comment:
I think the `helper` field might be the most appropriate here if you wanted to re-use one ? I named it a little poorly but the intention was to collect information about cases where `tbot` was being wrapped or invoked by some other tool.